docs/source/java/vector.rst

```diff
@@ -268,6 +268,82 @@ For example, the code below shows how to build a :class:`ListVector` of int's us
      }
   }
 
+Dictionary Encoding
+===================
+
+A :class:`FieldVector` can be dictionary encoded for performance or improved memory efficiency. While this is most often done with :class:`VarCharVector`, nearly any type of vector might be encoded if there are many values, but few unique values.
+
+There are a few steps involved in the encoding process:
+
+1. Create a regular, un-encoded vector and populate it
+2. Create a dictionary vector of the same type as the un-encoded vector. This vector must have the same values, but each unique value in the un-encoded vector need appear here only once.
+3. Create a :class:`Dictionary`. It will contain the dictionary vector, plus a :class:`DictionaryEncoding` object that holds the encoding's metadata and settings values.
+4. Create a :class:`DictionaryEncoder`.
+5. Call the encode() method on the :class:`DictionaryEncoder` to produce an encoded version of the original vector.
+6. (Optional) Call the decode() method on the encoded vector to re-create the original values.
+
+The encoded values will be integers. Depending on how many unique values you have, you can use either TinyIntVector, SmallIntVector, or IntVector to hold them. You specify the type when you create your :class:`DictionaryEncoding` instance. You might wonder where those integers come from: the dictionary vector is a regular vector, so the value's index position in that vector is used as its encoded value.
```

Comment:
Do we want to be consistent about linking classes?

FYI: `:class:` doesn't work for Java (I think these are failing or linking to C++?). I've been meaning to build up a workaround so it does work for Java.

Reply:
FWIW, I had no idea it was supposed to link. I've never worked with rst before. The highlighting is nice, though. 

I can add the class markup

Reply:
It's probably easiest to just use the double-backticks to put it in a monospace font, since what I'm saying is that the `:class:` markup won't actually link it properly since there's no integration between Javadoc and Sphinx

Reply:
(You should probably have seen build warnings to this effect)

Reply:
I was following the use of :class: in the other java tutorial pages (and elsewhere in this page).   I didn't see any warnings other than ones that seem to be about broken references, eg. 

/Users/lawrencewhite/Documents/GitHub/apache/arrow/docs/source/java/vector.rst:22: WARNING: unknown document: ../format/Columnar

Reply:
replaced :class: tags with double quotes

Reply:
Huh, I guess Sphinx doesn't complain in this case.